File: frontend/src/types/applicationTemplate/ApplicationTemplateUpdateDto.ts
import { CvConfigurationRequestDto } from '@/types/cv/CvConfigurationRequestDto.ts'
import { CoverLetterConfigurationUpdateDto } from '@/types/applicationTemplate/CoverLetterConfigurationUpdateDto.ts'

export interface ApplicationTemplateUpdateDto {
  id?: number
  name?: string
  cvConfiguration?: CvConfigurationRequestDto
  coverLetterConfiguration?: CoverLetterConfigurationUpdateDto
}

File: backend/src/main/kotlin/ch/streckeisen/mycv/backend/applicationTemplate/dto/ApplicationTemplateUpdateDto.kt
package ch.streckeisen.mycv.backend.applicationTemplate.dto

import ch.streckeisen.mycv.backend.cv.generator.CvConfigurationRequestDto

data class ApplicationTemplateUpdateDto(
    val id: Long?,
    val name: String?,
    val cvConfiguration: CvConfigurationRequestDto?,
    val coverLetterConfiguration: CoverLetterConfigurationUpdateDto?
)
